src/pages/dataElements/List.spec.tsx

```diff
@@ -0,0 +1,308 @@
+                            pageCount: 54,
+                        },
+                        result: [
+                            {
+                                ...filteredDataElementsMock.result[0],
+                                name: 'first page result',
+                            },
+                        ],
+                    }
+                    const secondPage = {
+                        pager: {
+                            page: 2,
+                            pageSize: 20,
+                            total: 1061,
+                            prevPage:
+                                '/dataElements/gist?pageListName=result&total=true&order=name:ASC&fields=access,id,name,code,domainType,valueType,lastUpdated,sharing&pageSize=20&page=1',
+                            nextPage:
+                                '/dataElements/gist?pageListName=result&total=true&order=name:ASC&fields=access,id,name,code,domainType,valueType,lastUpdated,sharing&pageSize=20&page=3',
+                            pageCount: 54,
+                        },
+                        result: [
+                            {
+                                ...filteredDataElementsMock.result[0],
+                                name: 'second page result',
+                            },
+                        ],
+                    }
+
+                    if (r.params.page === 1) {
+                        return Promise.resolve(firstPage)
+                    }
+                    if (r.params.page === 2) {
+                        return Promise.resolve(secondPage)
+                    }
+
+                    return Promise.reject('something wrong with mock')
+                },
+            }
+            return renderSection(dataProvider)
+        }
+        it('should display the page number', async () => {
+            const { getByTestId } = await renderWithPager()
+
+            const paginationFooter = getByTestId('section-list-pagination')
+
+            expect(paginationFooter).toHaveTextContent('Page 1 of 54')
+            expect(paginationFooter).toHaveTextContent('items 1-20 of 1061')
+        })
+        it('should disable previous page when on first page', async () => {
+            const { getByTestId } = await renderWithPager()
+            expect(
+                getByTestId('dhis2-uiwidgets-pagination-page-previous')
+            ).toHaveAttribute('disabled')
+        })
+        // next page
+        it('should allowing going to Next page', async () => {
+            const { getByTestId } = await renderWithPager()
+
+            userEvent.click(getByTestId('dhis2-uiwidgets-pagination-page-next'))
+
+            expect(pagerMock.mock.lastCall[0].params).toMatchObject(
+                expect.objectContaining({
+                    page: 2,
+                    pageListName: 'result',
+                    pageSize: 20,
+                    total: true,
+                })
+            )
```

Comment:
Here I think it'd be better to test

* for list items (there should be 20 and they should match what the api provided)
* that the displayed page number is "2"

The values that have been supplied to the query / api could be considered an implementation detail and it shouldn't matter what we do there as long as it works (which can be confirmed by looking at the list and pager UI). On the other hand, we don't know whether the correct items are displayed by looking at which values we send to the api

Reply:
In that case I think the description needs to be updated. It matches the description of the next test case (except that it's "next" instead of "previous", but the next test case actually checks the list's contents. Maybe sth like `should send the correct paging params to the endpoint`.

But I still think that, as long as the list renders correctly (like in the next test case), we don't need to test this. If we wanted to test this, I'd move this into a unit test which checks the hook (or whatever code it actually is) that send the parameters to the api